sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt

```diff
@@ -249,22 +262,30 @@ internal object KlaviyoApiClient : ApiClient {
                 val request = apiQueue.poll()
 
                 when (request?.send { broadcastApiRequest(request) }) {
+                    Status.Unsent -> {
+                        // Incomplete state: put it back on the queue and break out of serial queue
+                        apiQueue.offerFirst(request)
+                        break
+                    }
                     Status.Complete, Status.Failed -> {
                         // On success or absolute failure, remove from queue and persistent store
                         Registry.dataStore.clear(request.uuid)
+                        flushInterval = Registry.config.networkFlushIntervals[networkType].toLong()
```

Comment:
nit: Doesn't seem inherently clear to me why we are fetching the `flishInterval` in the `Status.Complete, Status.Failed` block when we don't use it till the `Status.PendingRetry` block

Reply:
Hm fair, I was just resetting in case of retries that had increased the timer. Wasn't even thinking about the network timers aspect of this